docs/setup/settings.asciidoc

```diff
@@ -36,11 +36,21 @@ Set to `false` to disable Console. *Default: `true`*
  <<ops-cGroupOverrides-cpuAcctPath, `ops.cGroupOverrides.cpuAcctPath`>>.
 
 | `csp.rules:`
- | A https://w3c.github.io/webappsec-csp/[content-security-policy] template
+ | deprecated:[7.14.0,"In 8.0 and later, this setting will no longer be supported."] 
```

Comment:
> Are we sure we want to break this in 8.0?

We're pretty sure we want to remove this, yes (unless there are blocking reasons not to do so). `8.0` seems the best target, but the implementation allow us to do it later, potentially.

>  I think if we want to get rid of csp.rules in 8.0 we need to support the other directives.

Yea, definitely. I don't think we need (or even want) all of them though? Obvious necessary inclusions (e.g for cdn) would be `img-src` and `font-src`, but do you know the list of the directives we should support?

Reply:
> Are we sure we want to break this in 8.0? Besides just adding extra sources to these three directives, there's a possibility that users might need to add other CSP directives. I think if we want to get rid of `csp.rules` in 8.0 we need to support the [other directives](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy).

It's hard to say, but potentially just the widely supported/commonly used ones:

* `connect-src`
* `default-src`
* `font-src`
* `frame-src`
* `img-src`
* `frame-ancestors` (note that this would conflict with the `server.securityResponseHeaders.disableEmbedding` config option)
* `report-uri`
* `report-to`

Reply:
@joshdover do you think it make sense to add them all in current PR, or should it be done as a follow-up?

I feel like that at least `frame-ancestors` will require some additional discussions

Reply:
I think it's fine if we do this as a follow up since `csp.rules` is still available and not yet removed. +1 on needing to discuss how we're going to handle `frame-ancestors`.

Reply:
Created https://github.com/elastic/kibana/issues/102322